validation/src/validation_service/mod.rs

```diff
@@ -344,22 +343,18 @@ impl<C, N, P, SP> ParachainValidationInstances<C, N, P, SP> where
 			max_block_data_size,
 		));
 
-		let (_drop_signal, exit) = exit_future::signal();
-
 		let router = self.network.communication_for(
 			table.clone(),
 			&validators,
-			exit.clone(),
```

Comment:
this change seems well beyond the scope of a CLI change. (after #788 these changes will be unnecessary anyway)

I think that removing the `exit` future here will lead to orphaned tasks if this is merged as-is - probably a memory leak.

Reply:
Ah no actually it's the opposite. This whole PR is because @bkchr pointed me out that polkadot couldn't spawn the task properly because of the update I did.

With the CLI refactoring there is no exit future anymore, there is a base implementation in substrate already. So this must go. I also added a test here at the end that shows things exist gracefully now.

Reply:
Sure, these futures are going to run until the node shuts down. That is not what we want - this exit future specifically is about scoping futures to the end of the consensus session. Why have you removed it?

Reply:
You are totally right. I've mistaken that code with the other exit stuff. I just pushed a Revert commit on network/ and validation/ (and the small part of it in service/src/lib.rs